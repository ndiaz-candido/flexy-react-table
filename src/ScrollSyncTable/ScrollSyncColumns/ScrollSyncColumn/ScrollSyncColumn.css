.scrollSyncColumn {
  background: #FFF;
  border-right: 1px solid gray;
  padding: 10px;
  width: 120px;
  height: 200px;
}
