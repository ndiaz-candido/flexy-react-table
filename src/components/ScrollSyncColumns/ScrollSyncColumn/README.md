```js
<ScrollSyncTable>
  <ScrollSyncColumns>
    <ScrollSyncColumn name="a">Header A</ScrollSyncColumn>
  </ScrollSyncColumns>
  <ScrollSyncRows>
    <ScrollSyncRow>
      <ScrollSyncCell column="a">ScrollSyncColumn</ScrollSyncCell>
    </ScrollSyncRow>
  </ScrollSyncRows>
</ScrollSyncTable>
```
