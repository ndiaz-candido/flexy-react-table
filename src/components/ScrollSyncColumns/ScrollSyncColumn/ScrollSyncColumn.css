.scrollSyncColumn {
  padding: 10px;
}
