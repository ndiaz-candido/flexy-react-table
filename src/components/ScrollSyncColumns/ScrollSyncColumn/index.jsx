import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { scrollSyncColumn } from './index.module.css';

const ScrollSyncColumn = ({ width, children, className }) => {
  const styles = {
    width,
    flex: `0 0 ${width}`,
  };
  const classNames = classnames(className, scrollSyncColumn);

  return (
    <div className={classNames} style={styles}>
      {children}
    </div>
  );
};

ScrollSyncColumn.propTypes = {
  width: PropTypes.string,
  children: PropTypes.any.isRequired,

  /** The class name to override the default styles */
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

ScrollSyncColumn.defaultProps = {
  width: '150px',
  className: ''
};

export default ScrollSyncColumn;
