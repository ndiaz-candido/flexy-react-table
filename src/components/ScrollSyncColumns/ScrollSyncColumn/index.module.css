.scrollSyncColumn {
  padding: 15px;
  text-align: center;
}
