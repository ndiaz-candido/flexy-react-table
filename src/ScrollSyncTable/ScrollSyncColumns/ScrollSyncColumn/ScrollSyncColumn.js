import React from 'react';
import './ScrollSyncColumn.css';

const ScrollSyncColumn = ({ width }) => {
  const styles = {
    width: width,
    flex: `0 0 ${width}`
  };

  return (<div className="scrollSyncColumn" style={styles}>Column</div>);
};

ScrollSyncColumn.defaultProps = {
  width: '100px'
};

export default ScrollSyncColumn;
