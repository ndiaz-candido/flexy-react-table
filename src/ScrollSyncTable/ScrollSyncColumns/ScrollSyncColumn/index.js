import ScrollSyncColumn from './ScrollSyncColumn';

export default ScrollSyncColumn;
