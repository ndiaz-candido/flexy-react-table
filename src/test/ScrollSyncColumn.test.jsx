import React from 'react';
import { shallow } from 'enzyme';
import ScrollSyncColumn from '../components/ScrollSyncColumns/ScrollSyncColumn';

describe('ScrollSyncColumn', () => {
  test('should render its default content', () => {
    const ScrollSyncColumnComponent = shallow(<ScrollSyncColumn />);

    expect(ScrollSyncColumnComponent).toMatchSnapshot();
  });

  test('should render the given children', () => {
    const ScrollSyncColumnComponent = shallow(
      <ScrollSyncColumn>
        <div>Child component</div>
      </ScrollSyncColumn>
    );

    expect(ScrollSyncColumnComponent).toMatchSnapshot();
  });

  test('should use the given width and class name', () => {
    const ScrollSyncColumnComponent = shallow(
      <ScrollSyncColumn width="200px" className="a-class-name" />
    );

    expect(ScrollSyncColumnComponent).toMatchSnapshot();
  });
});
