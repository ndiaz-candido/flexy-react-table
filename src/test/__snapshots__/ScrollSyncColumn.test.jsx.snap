// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollSyncColumn should render its default content 1`] = `
<div
  className=""
  style={
    Object {
      "flex": "0 0 150px",
      "width": "150px",
    }
  }
/>
`;

exports[`ScrollSyncColumn should render the given children 1`] = `
<div
  className=""
  style={
    Object {
      "flex": "0 0 150px",
      "width": "150px",
    }
  }
>
  <div>
    Child component
  </div>
</div>
`;

exports[`ScrollSyncColumn should use the given width and class name 1`] = `
<div
  className="a-class-name"
  style={
    Object {
      "flex": "0 0 200px",
      "width": "200px",
    }
  }
/>
`;
